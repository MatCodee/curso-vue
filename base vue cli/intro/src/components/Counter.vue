<template>
    <h2>{{ customMsg }}</h2>
    <p data-textid="counter" >{{counter}}</p>
    <button v-on:click="increment">increment</button>
    <button v-on:click="decrement">decrement</button>
</template>

<script>
export default {
    //name: "counter",
    props: {
        msg: String,
        initial_counter: {
            type: Number,
            default: 100,
            //required: true,
            validator(value){
                return value > 0
            }
        },
    },
    data() {
        return {
            counter: this.initial_counter,
        }
    },

    methods: {
      increment() { this.counter++; },
      decrement() { if(this.counter > 0) this.counter--; }  
    },
    computed: {
        customMsg() {
            return this.msg ? this.msg : "Counter";
        }
    }
}
</script>

<style scoped>
p {
    font-size: 40px;
}
</style>