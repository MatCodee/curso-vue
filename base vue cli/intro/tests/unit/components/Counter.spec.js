import { shallowMount } from "@vue/test-utils"
import Counter from "@/components/Counter.vue";

describe('Counter Component',() => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(Counter);
    })
    /*
    test('Debe de hacer match con el snapchot',()=>{
        const wrapper = shallowMount(Counter);
        expect(wrapper.html()).toMatchSnapshot();

    });
    */
   test('h2 debe tener el valor por defecto "Counter"',()=>{        
        expect(wrapper.find('h2').exists()).toBe(true);
        const test_h2 = wrapper.find('h2');

        //console.log(test_h2.text());

        expect(test_h2.text()).toBe('Counter')
   });
   test('El valor por defecto debe ser 100 en el p',() => {
        expect(wrapper.find('[data-textid="counter"]').text()).toBe('100');
   });

   test('Incrementar y decrementar el contador',async () => {
        const [increaseBtn,decreaseBtn] = wrapper.findAll('button');
        
        await increaseBtn.trigger('click');
        await increaseBtn.trigger('click');
        await increaseBtn.trigger('click');
        await decreaseBtn.trigger('click');
        await decreaseBtn.trigger('click');

        expect(wrapper.find('[data-textid="counter"]').text()).toBe('101');
   });

   test('Debe establece rle valor por defecto',() => {
        const { initial_counter } = wrapper.props();
        const value = wrapper.find('[data-textid="counter"]').text();
        expect(Number(value)).toBe(initial_counter);
   });
   test('Debe de mostrar la prop title',() => {
        const wrapper = shallowMount(Counter, {
            props: {
                msg: "Hola mundo",
            }
        });
        expect(wrapper.find('h2').text()).toBe("Hola mundo");
   });

});