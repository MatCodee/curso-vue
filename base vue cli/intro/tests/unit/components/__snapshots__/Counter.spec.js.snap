// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe de hacer match con el snapchot 1`] = `
<h2>no definido</h2>
<p>10</p><button>increment</button><button>decrement</button>
`;
